### PR TITLE
Fix unused result warnings in augeas

### LINKED ISSUE #798: Warning with -Wunused-result
```
/home/markus/Projekte/Elektra/libelektra/src/plugins/augeas/augeas.c: In function ‘keySetOrderMeta’:
/home/markus/Projekte/Elektra/libelektra/src/plugins/augeas/augeas.c:49:2: warning: ignoring return value of ‘asprintf’, declared with attribute warn_unused_result [-Wunused-result]
  asprintf (&buffer, "%d", order);
  ^
/home/markus/Projekte/Elektra/libelektra/src/plugins/augeas/augeas.c: In function ‘elektraAugeasOpen’:
/home/markus/Projekte/Elektra/libelektra/src/plugins/augeas/augeas.c:345:3: warning: ignoring return value of ‘asprintf’, declared with attribute warn_unused_result [-Wunused-result]
   asprintf (&errormessage, "Unable to initialize augeas: %s", aug_error_message (augeasHandle));
   ^
/home/markus/Projekte/Elektra/libelektra/src/plugins/augeas/augeas.c: In function ‘foreachAugeasNode’:
/home/markus/Projekte/Elektra/libelektra/src/plugins/augeas/augeas.c:201:2: warning: ignoring return value of ‘asprintf’, declared with attribute warn_unused_result [-Wunused-result]
  asprintf (&matchPath, "%s//*", treePath);
  ^
/home/markus/Projekte/Elektra/libelektra/src/plugins/augeas/augeas.c: In function ‘saveTree’:
/home/markus/Projekte/Elektra/libelektra/src/plugins/augeas/augeas.c:312:3: warning: ignoring return value of ‘asprintf’, declared with attribute warn_unused_result [-Wunused-result]
   asprintf (&nodeName, AUGEAS_TREE_ROOT "%s", (keyName (key) + prefixSize));
   ^
/home/markus/Projekte/Elektra/libelektra/src/plugins/augeas/augeas.c: In function ‘removeOrphan’:
/home/markus/Projekte/Elektra/libelektra/src/plugins/augeas/augeas.c:162:3: warning: ignoring return value of ‘asprintf’, declared with attribute warn_unused_result [-Wunused-result]
   asprintf (&nodeMatch, "%s/*", treePath);
   ^
Linking C shared module ../../../lib/libelektra-augeas.so
```

### PR DESCRIPTION
**Pull request by fberlakovich**
This PR should fix #798. Please note the use of goto for memory allocation error handling. I know it is not perfect, but neither are the other solutions.

### CODE REVIEW

File: src/plugins/augeas/augeas.c [hunk 1]
```diff
@@ -143,14 +148,18 @@ static int convertToKey (augeas * handle, const char * treePath, void * data)
 	/* fill key values */
 	keySetString (key, value);
 	conversionData->currentOrder++;
-	keySetOrderMeta (key, conversionData->currentOrder);
+	result = keySetOrderMeta (key, conversionData->currentOrder);
+
+	if (result < 0) return result;
```
Comment (markus2330) [inline src/plugins/augeas/augeas.c]: Is the dropping of the key here really what we want? Can you comment (in the source) what you intended here?

**Comment by markus2330** [comment]
Looks ok (I am unsure about the commented line). Please add some more code comments.

**Comment by fberlakovich** [comment]
I added comments to the functions used as function pointers. The plugin never drops only a single key, but fails completely in case of an error. I think this is the right thing to do as a leftout or leftover key might corrupt the resulting configuration file.

**Comment by markus2330** [comment]
Thank you for improving the quality of such error cases!

**Pull request merged**